clients/drcachesim/tests/raw2trace_unit_tests.cpp

```diff
@@ -367,6 +382,11 @@ test_marker_placement(void *drcontext)
     raw.push_back(make_marker(TRACE_MARKER_TYPE_FUNC_ID, 0));
     raw.push_back(make_marker(TRACE_MARKER_TYPE_FUNC_RETADDR, 4));
     raw.push_back(make_marker(TRACE_MARKER_TYPE_FUNC_ARG, 2));
+    raw.push_back(make_block(offs_load2, 1));
+    raw.push_back(make_marker(TRACE_MARKER_TYPE_FUNC_ID, 0));
+    raw.push_back(make_marker(TRACE_MARKER_TYPE_FUNC_RETADDR, 4));
+    raw.push_back(make_marker(TRACE_MARKER_TYPE_FUNC_ARG, 2));
+    raw.push_back(make_block(offs_jmp, 1));
```

Comment:
I don't think this test is actually testing #5656 since there's nothing for the branch to be delayed after so it will always be after the markers?  If you disable the fix does this test fail?  This does not look sufficient.

Reply:
Added a new function `test_delay_markers()` and updated the test case. Thought it would be cleaner to separate it but the common parts for the three test functions could shared. Will do that in a separate CL.